tests/unit/test_resolution_legacy_resolver.py

```diff
@@ -252,7 +252,7 @@ class NotWorkingFakeDist(FakeDist):
             def metadata(self) -> email.message.Message:
                 raise FileNotFoundError(metadata_name)
 
-        dist = make_fake_dist(klass=NotWorkingFakeDist)
+        dist = make_fake_dist(klass=NotWorkingFakeDist)  # type: ignore[type-abstract]
```

Comment:
Hmm, why does this warn about the abstract class?

Reply:
Similar to the above, NotWorkingFakeDist doesn't implement BaseDistribution. So make_fake_dist gives you an object that doesn't do what it says it should

Reply:
For my understanding, is there an actual runtime problem here, or are we simply not able to express what we're doing in terms that the type system can understand?

It feels like we shouldn't be trying to type check mock objects like this as if they are full-fledged versions. After all, that's the whole point of mock objects, surely? What do we hope to gain from the type annotations in this case?

Reply:
mypy is correct to complain here. If you do `dist.from_directory` or many of the other methods defined by the `BaseDistribution` protocol, you'll see that `dist` does not implement the Protocol it claims to implement. You could use a narrower Protocol to type `_check_dist_requires_python`, but I think you'd find that not worth it.

So instead we just say "get out of my way, I know I'm doing a thing that could be unsafe, but I don't care because it's a test". `# type: ignore` is as good a way to say this as any other, such as casting to Any. In general, type checking tests is lower value than type checking other code.

Reply:
… Continuing from the comment above, it should be OK to add a comment here (and one for FakeDistribution) to say we’re kind ot intentionally not fleshing out the implementation since an unimplemented abstract method would correctly raise and make a test fail if it’s accidentally called.